install.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+#
+# A basic install wrapper for UNIX-based systems.
+#
+
+GITHUB_CHANGES_FILE="https://raw.githubusercontent.com/a-scie/lift/main/CHANGES.md"
+GITHUB_DOWNLOAD_BASE="https://github.com/a-scie/lift/releases/download"
+
+# Check if curl is available.
+if ! command -v curl &> /dev/null; then
+  echo "Error: curl command not found. Please install curl and try again." >&2
+  exit 1
+fi
+
+# Check if arch is available.
+if ! command -v arch &> /dev/null; then
+  echo "Error: arch command not found. Please install arch and try again." >&2
+  exit 1
+fi
+
+# Map platform architecture to released binary architecture.
+READ_ARCH="$(arch)"
+case "$READ_ARCH" in
+  x86_64*)   ARCH="x86_64" ;;
+  arm64*)    ARCH="aarch64" ;;
+  aarch64*)  ARCH="aarch64" ;;
+  *)        echo "unknown arch: $(READ_ARCH)"; exit 1 ;;
+esac
+
+# Map OS name to released binary OS names and platform-specific SHA checker invocation.
+case "$OSTYPE" in
+  darwin*)  OS="macos" SHASUM_CMD="shasum -a 256 -c -" ;;
+  linux*)   OS="linux" SHASUM_CMD="sha256sum -c -" ;;
+  *)        echo "unknown platform: ${OSTYPE}"; exit 1 ;;
```

Comment:
That's disingenuous since we release for Windows! It would be more honest to say not supported, download from here, etc. at the least.

Reply:
punted on windows because supporting it is unfamiliar in the context of UNIX shell script execution to me (I'd assume a windows-mode installer would be e.g. `install.bat` or `install.ps1`), but I suppose the windows mode of support here is just "run in WSL" which behaves largely as UNIX.

Reply:
> but I suppose the windows mode of support here is just "run in WSL" which behaves largely as UNIX.

Nope. So, the reasoning goes like this:
1. If you use science, you are, by definition, a developer
2. Developers will almost always use either git or wsl or they'll have wsl installed for Docker use since its the most performant Docker backend on Windows.

Item 2 ensures bash. Git installs its own MSYS2 bash to even be able to work. Windows does likewise when it has wsl installed.

So you can almost always run bash from real Windows (not wsl) when you use Windows and are a developer. The scie-jump ITs, for example, run on Windows in CI and those are powered by an examples/run.sh bash script.
--
All that said, there is no excuse. I need to learn PowerShell and come up with a 1st class script. They release for Linux, Mac, Windows, so really no excuse: https://github.com/PowerShell/PowerShell/releases/tag/v7.4.5

Reply:
aha, but why port any logic to powershell interpreted script if powershell can just run bash natively?

Reply:
Curl did not work. I can't remember the details, but after alot of fail, using pwsh to do downloads worked.